rfd/0067-database-access-aws-redis

```diff
@@ -0,0 +1,221 @@
+    RandomValueLength int
+}
+
+// Secrets defines an interface for managing secrets. A secret consists of a
+// key path and a list of versions which hold copies of current or past secret
+// values.
+type Secrets interface {
+    // Create creates a new secret.
+    Create(req CreateRequest) (*SecretValue, error)
+
+    // Delete deletes the secret for provided path.
+    Delete(key string) error
+
+    // PutValue creates a new secret version for the secret. Implementations
+    // must guarantee that only the first of many calls to PutValue succeeds,
+    // when req.CurrentValue are the same.
+    PutValue(req PutValueRequest) (*SecretValue, error)
+
+    // GetValue returns the secret value for provided version. Besides version
+    // string returned from PutValue, two specials versions "CURRENT" and
+    // "PREVIOUS" can also be used to retrieve the current and previous
+    // versions respectively. If version is empty, "CURRENT" is used.
+    GetValue(key string, version string) (*SecretValue, error)
+}
+```
+It is up to the caller to call `PutValue` periodically to rotate the secret.
+
+#### Secret store - Random password generation
+[sethvargo/go-password](https://github.com/sethvargo/go-password) can be used to generate random passwords. <sup>[3]<sup>
+
+A minimum length of 16 should be enforced for generating random passwords. A mix of lower case letters, upper case letters, digits, and symobols should be also enforced.
+
+#### Secret store - AWS Secrets Manager
+With `iam:PutUserPolicy` and `iam:PutRolePolicy` permissions, Teleport service can grant required permissions to itself to perform Secrets Manager APIs. Users can limit the permissions by setting the following permissions boundaries (preferably using bootstrap configurator):
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": {
+    "Effect": "Allow",
+    "Action": [
+      "secretsmanager:CreateSecret",
+      "secretsmanager:GetSecretValue",
+      "secretsmanager:PutSecretValue",
+      "secretsmanager:DeleteSecret",
+    ],
+    "Resource": "arn:aws:secretsmanager:{region}:{account-id}:secret:{key-prefix}*"
+  }
+}
+```
+Here `key-prefix` is used to prevent Teleport from accessing other secrets in the AWS Secrets Manager.
+
+For racing `PutSecretValue` calls, [`PutSecretValueInput.ClientRequestToken`](https://docs.aws.amazon.com/sdk-for-go/api/service/secretsmanager/#PutSecretValueInput) can be used to "ensure idempotency" as AWS prevents different secret values of the same `ClientRequestToken` to be writen. When preparing `PutSecretValue` call, a MD5 UUID can be generated from version string of the CURRENT value, and used for `ClientRequestToken`. When `PutSecretValue succeeds, the `ClientRequestToken` becomes the version string of the latest value. This effectively makes version strings rolling MD5 UUIDs of their previous versions, and only the first call to create each version can succeed.
```

Comment:
Nit: Could you wrap the lines in the RFD to 80 characters so its easier to read w/o horizontal scrolling please? :)